resteasy-jaxrs/src/main/java/org/jboss/resteasy/annotations/jaxrs/QueryParam.java

```diff
@@ -0,0 +1,16 @@
+package org.jboss.resteasy.annotations.jaxrs;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Retention(RUNTIME)
+@Target({ FIELD, METHOD, PARAMETER })
+public @interface QueryParam
+{
+   String value() default "";
```

Comment:
@asoldano I assume that value of this annotation should behave like jax-rs QueryParam annotation (behaviour is not described in javadoc or jira). Anyway, I tried to use this parameter:

`@QueryParam("queryParam3") String queryParam3Updated`

And RESTEasy uses value of `queryParam3Updated` HTTP parameter. I guess that RESTEasy should use value of `queryParam3` HTTP parameter, right?

Reply:
Yes, that's my understanding too.

Reply:
+1.
@marekkopecky , maybe you can provide a test on this too as part of the tests you're working on? Then we (I) have a look :)

Reply:
@asoldano Sure. I will prepare PR tomorrow or on Monday ...

Reply:
@asoldano I prepare tests in [my branch](https://github.com/marekkopecky/Resteasy/commits/RESTEASY-1880-master-tests) for master upstream branch. Custom values of new annotations are tested [here](https://github.com/marekkopecky/Resteasy/commit/ea872100cb47d1ab567a61b52bc8eb989ec2215e#diff-2923ffdb33f3ce8e5e45ef599d048b96R133). If you want, I can prepare PR or prepare branch with tests for 3.6.

Reply:
https://github.com/resteasy/Resteasy/pull/1510
RESTEasyParamBasicTest#customValuesTest is in  ExpectedFailing category in this PR ...